src/main/java/com/google/devtools/build/lib/sandbox/cgroups/controller/Controller.java

```diff
@@ -0,0 +1,37 @@
+package com.google.devtools.build.lib.sandbox.cgroups.controller;
+
+import com.google.common.base.Defaults;
+import com.google.common.base.Suppliers;
+import com.google.devtools.build.lib.events.Event;
+import com.google.devtools.build.lib.events.EventHandler;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.nio.file.Path;
+import java.util.function.Supplier;
+
+public interface Controller {
+    default boolean isLegacy() throws IOException {
+        return !getPath().resolve("cgroup.controllers").toFile().exists();
```

Comment:
Do we need to make a filesystem call here everytime we want to check this method? Since the controllers are already specialized into `Legacy` and `Unified`, can we just override this there with `true` and `false`?

Reply:
Good call out, updated